util/index.go

```diff
@@ -16,3 +16,14 @@ func ValidateHeightIndexed(reader archive.Reader, height uint64) error {
 	}
 	return nil
 }
+
+func ValidateRegisterHeightIndexed(reader archive.Reader, height uint64) error {
+	h, err := reader.LatestRegisterHeight()
+	if err != nil {
+		return fmt.Errorf("could not get last indexed register height for Archive node")
+	}
+	if height > h {
```

Comment:
what happens if `height` is higher than the first block indexed?

Reply:
Good catch, this mirrors the logic for `ValidateHeightDataAvailable` up above, but I've updated both functions to reflect the range check on both bounds